core/common/src/main/scala/org/typelevel/otel4s/context/LocalProvider.scala

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.context
+
+import cats.Applicative
+import cats.effect.IOLocal
+import cats.effect.LiftIO
+import cats.effect.MonadCancelThrow
+import cats.mtl.Local
+import org.typelevel.otel4s.instances.local._
+
+/** A utility class to simplify the creation of the [[cats.mtl.Local Local]].
+  *
+  * @tparam F
+  *   the higher-kinded type of a polymorphic effect
+  *
+  * @tparam Ctx
+  *   the type of the context
+  */
+@annotation.implicitNotFound("""
+Cannot find the `LocalProvider` for effect `${F}` and context `${Ctx}`.
+The `LocalProvider` can be derived implicitly:
+
+1) from implicit `IOLocal[${Ctx}]` and `LiftIO[${F}]`:
+
+IOLocal(${Ctx}.root).map { implicit ioLocal =>
+  val provider = LocalProvider[IO, ${Ctx}]
+}
+
+2) from implicit `Local[${F}, ${Ctx}]`:
+
+implicit val local: Local[${F}, ${Ctx}] = ???
+val provider = LocalProvider[${F}, ${Ctx}]
+
+3) from implicit `LiftIO[${F}]`:
+
+val provider = LocalProvider[IO, ${Ctx}]
+""")
+trait LocalProvider[F[_], Ctx] {
+
+  /** Creates a [[cats.mtl.Local Local]] instance. The method is invoked once
+    * per creation of the Otel4s instance.
+    */
+  def local: F[Local[F, Ctx]]
+}
+
+object LocalProvider extends LocalProviderLowPriority {
+
+  def apply[F[_], C](implicit ev: LocalProvider[F, C]): LocalProvider[F, C] = ev
+
+  /** Creates [[LocalProvider]] that derives [[cats.mtl.Local Local]] instance
+    * from the given [[cats.effect.IOLocal IOLocal]].
+    *
+    * @param ioLocal
+    *   the [[cats.effect.IOLocal IOLocal]] to use
+    *
+    * @tparam F
+    *   the higher-kinded type of a polymorphic effect
+    *
+    * @tparam Ctx
+    *   the type of the context
+    */
+  def fromIOLocal[F[_]: MonadCancelThrow: LiftIO, Ctx](
+      ioLocal: IOLocal[Ctx]
+  ): LocalProvider[F, Ctx] =
+    new LocalProvider[F, Ctx] {
+      val local: F[Local[F, Ctx]] =
+        MonadCancelThrow[F].pure(
+          localForIOLocal[F, Ctx](implicitly, implicitly, ioLocal)
```

Comment:
The `ioLocal`  argument (of the top-level function) is not implicit and the compiler cannot derive the instance 🙁 :

```
[error] /.../otel4s/core/common/src/main/scala/org/typelevel/otel4s/context/LocalProvider.scala:83:11: Could not find an implicit instance of Local[F, Ctx]. If you have a
[error] value of type Ctx in scope, or a way of computing one, you may want to construct
[error] a value of type Kleisli for this call-site, rather than type F. An example type:
```